src/qca/utils/utils.py

```diff
@@ -129,6 +131,14 @@ def get_T_depth(cpt_circuit: cirq.AbstractCircuit):
                 break
     return t_depth
 
+def complete_decomposition(circuit: cirq.AbstractCircuit) -> cirq.AbstractCircuit:
+    prev_decomp = None
+    context = cirq.DecompositionContext(gam.gam)
+    while circuit != prev_decomp:
+        prev_decomp = circuit
+        circuit = circuit_decompose_multi(circuit, 1, context=context)
```

Comment:
Only thing here, why is the 1 hardcoded for N in the circuit_decompose_multi function, since it represents the max_decomposition_passes wanted, shouldn't it be an argument of the function? I see in PyLIQTR's uses, they toggle between 1 and 2, but have not seen anything more, so just wanted to make sure of this.

Reply:
There is no way for the user to be able to accurately know the number of rounds of decomposition are necessary ahead. An Rx decomposition may take only one round, but a complete qubitized phase estimation circuit make take about 6 rounds or so of decomposition until we get everything in their single qubit correspondents. This function enforces that we keep decomposing until we have done so, which is why you see the comparison being done so. Once `circuit == prev_decomp` its because we have completely decomposed the circuit

Reply:
Oh I see, this makes sense!